### PR TITLE
Query Panels: Pass on loading state

### LINKED ISSUE #200: Loading spinner in query editor gets stuck after the query is finished
**What happened**:
When editing a query in the editor, if the query is invalid and returns an error, after which another is run, the second query will resolve, but the Run button will still have the loading spinner. N.B. This only happens if the second query also returns an error. 

**What you expected to happen**:
The loading spinner should disappear when a query is resolved. 

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a new panel and choose Redshift as DS
2. Select tables and enter an invalid query then click Run
3. Once the query resolves and errors out, edit it to another invalid query (or rerun the first one) and click Run

**Screenshots**
ezgif-5-3d0ab25ab9

**Anything else we need to know?**:
Maybe related to  [this](https://github.com/grafana/redshift-datasource/issues/188)
**Environment**:
- Grafana version: 9.1.2
- Plugin version: 1.4.1
- OS Grafana is installed on:
- User OS & Browser:
- Others:

### PR DESCRIPTION
**Pull request by idastambuk**
Fixes [Redshift #200](https://github.com/grafana/redshift-datasource/issues/200)

A bit of background [here](https://github.com/grafana/grafana/pull/61635)

Decided to go with the quick fix as assessing the impact of clearing error state would take more time, and knowledge, than I currently have. 
The fix makes sure the loading state is passed to the query editors and not overwritten. This way query editors always have the most updated loading state.

N.B. This doesn't fix the annotations editor, but the buttons shouldn't be there anyway, so [this PR](https://github.com/grafana/redshift-datasource/pull/206) hides them. 

  If anyone that has background knowledge on preProcessPanelData and its impact, wants to take a look, you’re welcome to do so and Im available for questions.

### CODE REVIEW

File: public/app/features/query/components/QueryEditorRow.test.ts
Comment (sarahzinger) [inline public/app/features/query/components/QueryEditorRow.test.ts:127]: It's very possible I'm misreading this, but I think there's nothing about this test that ensures this is an old error vs a new error. All this is testing is regardless of whether the error happened in this panel, or another panel, if not all the queries have completed keep the state as `Loading`, right? To phrase this another way: would we still expect if Panel A fails while Panel B is still loading, should the state be set to "error" or should it be set to "loading"? This change is suggesting it should be set to "loading", which I think is a reasonable change, after all you may want to "cancel" your query and can't cancel something that is in error. 

That change will indirectly also help your bug where old errors are being kept in state (most likely in an attempt to prevent errors from "flashing" when a user refreshes a still broken query).

I wonder if it would make sense to keep both the test above and the test below this but swap the order and then rename them to something like
- should keep the panel state to Loading if it's still loading other panels and this panel has errored
- should keep the panel state to Loading if it's still loading other panels and another panel has errored

Or something like that? Words are tricky.

**Review by sarahzinger — APPROVED** [review]
@idastambuk a quick test name clarification, but otherwise seems ok to me! That said maybe we should chat about this in frontend weekly before merging just to make sure there aren't other implications we're not thinking about

File: public/app/features/query/components/QueryEditorRow.test.ts
Comment (idastambuk) [reply public/app/features/query/components/QueryEditorRow.test.ts:127]: You're right, thinking about it, I guess the test is really testing the current logic and not JUST the expected behavior from the user's standpoint. That is, the only way panelData A would be in this state is if it returned an error previously, but you can only know that if you follow the query result flow. This test can't ensure that we're getting the status in sequential order. 
I would say we expect both A and B to get the loading state, since both queries are run, no matter which "Run" button the user clicks. I think your suggestion to rename them makes sense then.

Comment (idastambuk) [reply public/app/features/query/components/QueryEditorRow.test.ts:127]: Renamed them to 
`
it(should keep the state in loading until all queries are finished, even if the current query has errored)
it(should keep the state in loading until all queries are finished, if another query has errored)
`
LMK if that sounds good

**Comment by sarahzinger** [comment]
I think this makes a lot of sense to me and seems like a great fit for 9.5! But again would be nice if we can get another set of eyes if possible from someone more familiar or with more ownership over this experience. @grafana/dashboards-squad or maybe @grafana/user-essentials any ideas? (apologies I'm not sure who owns the error and loading handling experience of panels?)We're looking to update the loading/error states as they are passed to plugins from query panels.

We think the bug is this:
- user makes a request with multiple queries in a panel.
- one of the panels fails
- which sets the loading state for all queries to go from loading to error. (First Question: Is this intentional? I could see this being a deliberate choice, maybe it helps to surface errors as quickly as possible to users, but it puts us in a difficult spot when we have long running queries that the user may want to cancel, but can not because we are no longer showing the choice to cancel a loading query and instead are showing an error state, even as other queries run in the background)
- then the user refreshes this same set of queries that previously failed. While the queries loads a second time, the previous error is not removed in state. (Second question, is this intentional? I could see this being done intentionally so as to prevent an error from clearing before, but it also is a bit confusing for plugin developers?)
- then because the previous state is in an error, we immediately set the state to error again, before ever going to a loading state. That error state is cleared only if the query finishes loading successfully, never really showing loading. This is challenging for us when making long running queries because [we assume](https://github.com/grafana/grafana-async-query-data-js/blob/main/src/RunQueryButtons.tsx#L21) that loading state will only ever go from error -> loading -> error, not error -> error, and end up showing a loading spinner forever rather than resolving to the new error. 

Our proposal here is to ensure that even if the last response's error is sent along to the query panel, we should always pass along a loading state if it is loading to the query editors in case they need to show that loading state. Does this seem reasonable?

**Review by torkelo — APPROVED** [review]

**Pull request merged**
